flink/src/test/java/org/apache/iceberg/flink/TestFlinkFilters.java

```diff
@@ -0,0 +1,361 @@
+
+    Assert.assertEquals(expected.op(), not.op());
+    assertPredicatesMatch((UnboundPredicate<?>) expected.child(), (UnboundPredicate<?>) not.child());
+  }
+
+  @Test
+  public void testLike() {
+    UnboundPredicate<?> expected = org.apache.iceberg.expressions.Expressions.startsWith("field5", "abc");
+    Expression expr = resolve(ApiExpressionUtils.unresolvedCall(
+        BuiltInFunctionDefinitions.LIKE, Expressions.$("field5"), Expressions.lit("abc%")));
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+    assertPredicatesMatch(expected, (UnboundPredicate<?>) actual.get());
+  }
+
+  private void matchLiteral(String fieldName, Object flinkLiteral, Object icebergLiteral) {
+    Expression expr = resolve(Expressions.$(fieldName).isEqual(Expressions.lit(flinkLiteral)));
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+
+    BoundLiteralPredicate predicate =
+        (BoundLiteralPredicate<?>) ((UnboundPredicate<?>) actual.get())
+            .bind(FlinkSchemaUtil.convert(TABLE_SCHEMA).asStruct(), false);
+    Assert.assertTrue(predicate.test(icebergLiteral));
+  }
+
+  private static Expression resolve(Expression originalExpression) {
+    return originalExpression.accept(new ApiExpressionDefaultVisitor<Expression>() {
+      @Override
+      public Expression visit(UnresolvedReferenceExpression unresolvedReference) {
+        String name = unresolvedReference.getName();
+        Optional<TableColumn> field = TABLE_SCHEMA.getTableColumn(name);
+        if (field.isPresent()) {
+          int index = TABLE_SCHEMA.getTableColumns().indexOf(field.get());
+          return new FieldReferenceExpression(name, field.get().getType(), 0, index);
+        } else {
+          return null;
+        }
+      }
+
+      @Override
+      public Expression visit(UnresolvedCallExpression unresolvedCall) {
+        List<ResolvedExpression> children =
+            unresolvedCall.getChildren().stream().map(e -> (ResolvedExpression) e.accept(this))
+                .collect(Collectors.toList());
+        return new CallExpression(unresolvedCall.getFunctionDefinition(), children, DataTypes.STRING());
+      }
+
+      @Override
+      public Expression visit(ValueLiteralExpression valueLiteral) {
+        return valueLiteral;
+      }
+
+      @Override
+      protected Expression defaultMethod(Expression expression) {
+        throw new UnsupportedOperationException(String.format("unsupported expression: %s", expression));
+      }
+    });
+  }
+
+  private void assertPredicatesMatch(UnboundPredicate<?> expected, UnboundPredicate<?> actual) {
+    Assert.assertEquals(expected.op(), actual.op());
+    Assert.assertEquals(expected.literal(), actual.literal());
+    Assert.assertEquals(expected.ref().name(), actual.ref().name());
```

Comment:
These assertions also need context strings to identify which part of the predicate is tested, like "Predicate operation should match"